contracts/factory/src/contract.rs

```diff
@@ -43,30 +43,30 @@ pub fn instantiate(
         token_code_id: msg.token_code_id,
         fee_address: None,
         generator_address: deps.api.addr_validate(msg.generator_address.as_str())?,
-        pair_xyk_config: None,
-        pair_stable_config: None,
     };
 
     if let Some(fee_address) = msg.fee_address {
         config.fee_address = Some(deps.api.addr_validate(fee_address.as_str())?);
     }
 
-    if let Some(pair_xyk_config) = msg.pair_xyk_config {
-        if pair_xyk_config.valid_fee_bps() {
-            config.pair_xyk_config = Some(pair_xyk_config);
-        } else {
-            return Err(ContractError::PairConfigInvalidFeeBps {});
-        }
+    let config_set: HashSet<String> = msg
+        .pair_configs
+        .clone()
+        .into_iter()
+        .map(|pc| pc.pair_type.to_string())
+        .collect();
```

Comment:
May be it's better to replace ".clone().into_iter()" with ".iter()"?

Reply:
fixed